Src/Web/Web.Shared.Net/RequestTrackingTelemetryModule.cs

```diff
@@ -309,6 +323,40 @@ internal void OverrideCorrelationIdLookupHelper(CorrelationIdLookupHelper correl
         }
 
         /// <summary>
+        /// A request must be tracked as Active in order for telemetry to be recorded within OnEndRequest().
+        /// <see cref="System.Web.Handlers.TransferRequestHandler">
+        /// TransferRequestHandler can create a Child request to route extension-less requests to a controller.
+        /// (ex: site/home -> site/HomeController.cs)
+        /// We do not want duplicate telemetry logged for both the Parent and Child requests, so the activeRequests will be created OnBeginRequest.
+        /// When the child request OnEndRequest, the id will be removed from this dictionary and telemetry will not be logged for the parent.
+        /// </see>
+        /// </summary>
+        /// <remarks>
+        /// Unit test projects cannot create an [internal] IIS7WorkerRequest object.
+        /// Without this object, we cannot modify the Request.Headers without throwing a PlatformNotSupportedException.
+        /// Unit tests will have to initialize the RequestIdHeader.
+        /// The second IF will ensure the id is added to the activeRequests.
+        /// </remarks>
+        private void EnsureIsActiveRequest(HttpContext context)
+        {
+            string requestId;
+            if (context.Request.Headers[RequestIdHeader] == null)
+            {
+                requestId = Guid.NewGuid().ToString();
+                context.Request.Headers[RequestIdHeader] = requestId;
+            }
+            else
+            {
+                requestId = context.Request.Headers[RequestIdHeader];
+            }
+            
+            if (!activeRequests.ContainsKey(requestId))
```

Comment:
talked to @iusafaro. If this is for the test purposes only - I'd suggest to expose a protected Boolean flag that allows to disable this functionality. In reality such flag may be useful even as a public one (at least for the beta release) as we do know know if every version will allow to write to the header,